lsp-protocol.el

```diff
@@ -0,0 +1,419 @@
+;;; lsp-protocol.el --- Language Sever Protocol Bindings  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2020  Ivan Yonchovski
+
+;; Author: Ivan Yonchovski <yyoncho@gmail.com>
+;; Keywords: abbrev, abbrev, abbrev, abbrev
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Autogenerated bindings from lsp4j using
+;; https://github.com/victools/jsonschema-generator+scrips to generate
+;; scripts/generated.protocol.schema.json and then
+;; scripts/lsp-generate-bindings.el
+
+;;; Code:
+
+
+(defmacro lsp-interface (&rest interfaces)
+  "Generates LSP bindings from (Inteface name, Required Fields, Optional Fields) triplet.
+
+Example usage with dash.
+
+(-let [(&ApplyWorkspaceEditResponse
+  :failure-reason?) (ht (\"failureReason\" \"...\"))]
+  failure-reason?) "
+  (->> interfaces
+       (-partition 3)
```

Comment:
Can we generate list triplet instead of flatten list? This will save a `-partition` call. And easier to see the generated code.

Reply:
Updated, not sure if  the result is better.